metrics/resource_view_test.go

```diff
@@ -566,25 +582,38 @@ func TestStackDriverExports(t *testing.T) {
 				notReadyPodCountM.M(int64(3)))
 
 			records := []metricExtract{}
-			for record := range sdFake.published {
-				for _, ts := range record.TimeSeries {
-					records = append(records, metricExtract{
-						Name:   ts.Metric.Type,
-						Labels: ts.Resource.Labels,
-						Value:  ts.Points[0].Value.GetInt64Value(),
-					})
-					if strings.HasPrefix(ts.Metric.Type, "knative.dev/") {
-						if diff := cmp.Diff(ts.Resource.Type, metricskey.ResourceTypeKnativeRevision); diff != "" {
-							t.Errorf("Incorrect resource type for %q: (-want +got): %s", ts.Metric.Type, diff)
+		loop:
+			for {
+				select {
+				case record := <-sdFake.published:
+					for _, ts := range record.TimeSeries {
+						extracted := metricExtract{
+							Name:   ts.Metric.Type,
+							Labels: ts.Resource.Labels,
+							Value:  ts.Points[0].Value.GetInt64Value(),
+						}
+						// Override 'cluster-name' label to reset to a fixed value
+						if extracted.Labels["cluster_name"] != "" {
+							extracted.Labels["cluster_name"] = "test-cluster"
+						}
+						records = append(records, extracted)
+						if strings.HasPrefix(ts.Metric.Type, "knative.dev/") {
+							if diff := cmp.Diff(ts.Resource.Type, metricskey.ResourceTypeKnativeRevision); diff != "" {
+								t.Errorf("Incorrect resource type for %q: (-want +got): %s", ts.Metric.Type, diff)
+							}
 						}
 					}
-				}
-				if len(records) >= 2 {
-					// There's no way to synchronize on the internal timer used
-					// by metricsexport.IntervalReader, so shut down the
-					// exporter after the first report cycle.
-					FlushExporter()
-					sdFake.srv.GracefulStop()
+					if len(records) >= len(tc.expected) {
+						// There's no way to synchronize on the internal timer used
+						// by metricsexport.IntervalReader, so shut down the
+						// exporter after the first report cycle.
+						FlushExporter()
+						sdFake.srv.GracefulStop()
+						break loop
+					}
+				case <-time.After(5 * time.Second):
```

Comment:
Should those timeouts be consistent? it's 4s above.

Reply:
There used to be an extra delay on the Stackdriver outputs, but they should be similar now.